### PR TITLE
add auth_type param to fix issue 1789

### PR DESCRIPTION
**Pull request by daalvand**
I added   auth_type option to resolve the problem of duplicate requests in [1789 issue](https://github.com/ruflin/Elastica/issues/1789 "1789 issue")
```php
        'auth_type' => null //basic, digest, gssnegotiate, ntlm
```

### CODE REVIEW

File: src/Transport/Http.php [hunk 1]
```diff
@@ -222,4 +222,22 @@ protected function _getConnection(bool $persistent = true)
...
+                break;
+            case "gssnegotiate":
+                return CURLAUTH_GSSNEGOTIATE;
+                break;
+            case "ntlm":
+                return CURLAUTH_NTLM;
+                break;
+            case "basic":
+            default:
+                return CURLAUTH_BASIC;
```
Comment (ruflin) [inline src/Transport/Http.php:240]: Before the default value was `CURLAUTH_ANY`. I wonder if this change could break some setups?

Comment (mehdiDalvand) [reply src/Transport/Http.php:239]: I changed the code and now the _getAuthType function returns CURLAUTH_ANY by default.

Comment (mehdiDalvand) [reply src/Transport/Http.php:239]: I added changelogs

**Comment by ruflin** [comment]
Overall the changes LGTM but it seems like some of the tests need to be updated as CI is failing: https://travis-ci.org/github/ruflin/Elastica/jobs/713737332

**Comment by daalvand** [comment]
I fixed the test failures but I do not know why still test has errors
[https://travis-ci.org/github/ruflin/Elastica/jobs/715142063](url)

**Comment by ruflin** [comment]
Thanks for updating the tests. The errors you see now are around linting. Can you run `make fix-phpcs` and check if there is a diff?

**Comment by daalvand** [comment]
The error was fixed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,12 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased](https://github.com/ruflin/Elastica/compare/7.0.0...master)
 ### Backward Compatibility Breaks
 ### Added
-
+Ability to specify the type of authentication manually by the `auth_type` parameter (in the client class config) was added (allowed values are `basic, digest, gssnegotiate, ntlm`)
```
Comment (ruflin) [inline CHANGELOG.md:10]: Nit: Sorry, didn't spot this before, but could you prefix it by `* ` to make it a list? Same for line 16.

Comment (daalvand) [reply CHANGELOG.md]: Oh you are right
I changed it.

**Comment by ruflin** [comment]
@daalvand Thank you very much for the contribution and going through the iterations!

**Pull request merged**